Assets/MixedRealityToolkit/Utilities/CameraFOVChecker.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit
+{
+    /// <summary>
+    /// Camera extension methods to test if colliders are within an FOV cone. Uses
+    /// caching to improve performance and ensure values are only computed once per frame
+    /// </summary>
+    public static class CameraFOVChecker
+    {
+        
+        // Help to clear caches when new frame runs
+        static private int inFOVConeLastCalculatedFrame = -1;
+        // Map from grabbable => is the grabbable in FOV for this frame. Cleared every frame
+        private static Dictionary<(Collider collider, Camera camera), bool> inFOVConeColliderCache = new Dictionary<(Collider collider, Camera camera), bool>();
+        // List of corners shared across all sphere pointer query instances --
+        // used to store list of corners for a bounds. Shared and static
+        // to avoid allocating memory each frame
+        private static List<Vector3> inFOVConeBoundsCornerPoints = new List<Vector3>();
+
+        /// <summary>
+        /// Returns true if a collider's bounds is within the camera FOV. 
+        /// Utilizes a cache to test if this collider has been seen before and returns current frame's calculated result.
+        /// </summary>
+        /// <param name="myCollider">The collider to test</param>
+        public static bool IsInFOVConeCached(this Camera cam,
+            Collider myCollider)
+        {
+            // if the collider's size is zero, it is not visible. Return false.
+            if(myCollider.bounds.size == Vector3.zero || myCollider.transform.localScale == Vector3.zero)
```

Comment:
```suggestion
            if (myCollider.bounds.size == Vector3.zero || myCollider.transform.localScale == Vector3.zero)
```

Reply:
btw, should these be floating point margin of error comparison instead of exact comparison to zero?

Reply:
I assumed no, since people might want to set something invisible by setting it exactly to Vector3.zero. Somebody might have a collider whose scale is very small but not zero, to represent say a point. Didn't want to ignore that.